### DISCUSSION TITLE
Accept --max-http-header-size as a valid flag

### DISCUSSION THREAD

**Discussion by shivjisakina** [vercel/next.js#14340] [Help]
# Feature request

Be able to add --max-http-header-size to increase the maximum header size allowed passed 8k that is built in with Node.js

## Is your feature request related to a problem? Please describe.

After the Node.js update that decreased the max http size down from 80k to 8k we're trying to allow a higher header size so users arent dealing with the 431 error

## Describe alternatives you've considered

We tried using cross-env and it works locally, but once deployed to the DEV environment (using Azure), the flag is ignored or it throws an "export: --max-http-header-size bad variable name" error

**Comment by paambaati** [comment]
You should be able to set it via the [`NODE_OPTIONS`](https://nodejs.org/api/cli.html) environment variable.

Use —

```
NODE_OPTIONS='--max-http-headers-size=8'
```

**Reply by balazsorban44** [reply]
Looks like it has been changed to 16k by default since Node `v13.13.0`
https://nodejs.org/api/cli.html#cli_max_http_header_size_size

So ~I guess~ (tested it) **updating** to something newer **will fix** this as well.
